network/gossip.go

```diff
@@ -12,21 +13,23 @@ import (
 )
 
 type gossipService struct {
-	ctx     context.Context
-	wg      sync.WaitGroup
-	host    lp2phost.Host
-	pubsub  *lp2pps.PubSub
-	topics  []*lp2pps.Topic
-	subs    []*lp2pps.Subscription
-	eventCh chan Event
-	logger  *logger.SubLogger
-
-	generalTopicName string
-	rateLimit        *rateLimit
+	ctx                    context.Context
+	wg                     sync.WaitGroup
+	config                 *Config
+	host                   lp2phost.Host
+	pubsub                 *lp2pps.PubSub
+	topics                 []*lp2pps.Topic
+	subs                   []*lp2pps.Subscription
+	generalTopicDeprecated *lp2pps.Topic
```

Comment:
Instead rename topic to deprecated you can use standard comment for deprecated, for example:

```go
generalTopic *lp2pps.Topic // Deprecated: generalTopic ...
```

Reply:
Done.